activetable/activetable.py

```diff
@@ -0,0 +1,297 @@
+        default='ActiveTable problem'
+    )
+    content = String(
+        display_name='Table definition',
+        help='The definition of the table in Python-like syntax.',
+        scope=Scope.content,
+        multiline_editor=True,
+        resettable_editor=False,
+        default=textwrap.dedent("""\
+        [
+            ['Column header 1', 'Column header 2'],
+            ['Enter "answer" here:', String(answer='answer')],
+            [42, Numeric(answer=42, tolerance=0.0)],
+        ]
+        """)
+    )
+    help_text = String(
+        display_name='Help text',
+        help='The text that gets displayed when clicking the "+help" button.  If you remove the '
+        'help text, the help feature is disabled.',
+        scope=Scope.content,
+        multiline_editor=True,
+        resettable_editor=False,
+        default='Fill in the cells highlighted in yellow with the correct answers.  '
+        'When you are done, you can check your answers using the button below the table.',
+    )
+    column_widths = String(
+        display_name='Column widths',
+        help='Set the width of the columns in pixels.  The value should be a Python-like list of '
+        'numerical values.  The total width of the table should not be more than 800. No value '
+        'will result in equal-width columns with a total width of 800 pixels.',
+        scope=Scope.content,
+        resettable_editor=False,
+    )
+    row_heights = String(
+        display_name='Row heights',
+        help='Set the heights of the rows in pixels.  The value should be a Python-like list of '
+        'numerical values. Rows may grow higher than the specified value if the text in some cells '
+        'in the row is long enough to get wrapped in more than one line.',
+        scope=Scope.content,
+        resettable_editor=False,
+    )
+    default_tolerance = Float(
+        display_name='Default tolerance',
+        help='The tolerance in pecent that is used for numerical response cells you did not '
+        'specify an explicit tolerance for.',
+        scope=Scope.content,
+        default=1.0,
+    )
+    max_score = Float(
+        display_name='Maximum score',
+        help='The number of points students will be awarded when solving all fields correctly.  '
+        'For partially correct attempts, the score will be pro-rated.',
+        scope=Scope.settings,
+        default=1.0,
+    )
+    max_attempts = Integer(
+        display_name='Maximum attempts',
+        help='Defines the number of times a student can try to answer this problem.  If the value '
+        'is not set, infinite attempts are allowed.',
+        scope=Scope.settings,
+    )
+
+    editable_fields = [
+        'display_name',
+        'content',
+        'help_text',
+        'column_widths',
+        'row_heights',
+        'default_tolerance',
+        'max_score',
+        'max_attempts',
+    ]
+
+    # Dictionary mapping cell ids to the student answers.
```

Comment:
Are cell IDs stable while editing? Inserting a row in the table will make students' existing answers mismatched, right?  Is this understood by the course team? Can we say something in the help text?